packages/expo-modules-core/ios/AppDelegates/EXAppDelegateWrapper.h

```diff
@@ -6,10 +6,24 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface RCTAppDelegate ()
+
+- (RCTRootViewFactory *)createRCTRootViewFactory;
```

Comment:
is that necessary? i doubt if that will pollute all code referencing RCTAppDelegate, because this is inside header (not .m or .mm).

Reply:
It is necessary. `createRCTRootViewFactory` is not public, so we couldn't override it in Swift. I'll try to do it another way, but I think we should change it upstream.

Reply:
maybe just remove the `override` at https://github.com/expo/expo/blob/45fe2098524a5575c5e44e9c105b01a6fe671640/packages/expo-modules-core/ios/AppDelegates/ExpoAppInstance.swift#L16 😅

Reply:
As far as I know, then that function will not be called (Objective-C will not be aware that there is a function in the extending Swift class to call). Also, if this function will become public at some point, that would be a breaking change for us (you have to add `override` back).

Reply:
haha correct. i was trying from the legacy EXAppDelegateWrapper: https://github.com/expo/expo/blob/45fe2098524a5575c5e44e9c105b01a6fe671640/packages/expo-modules-core/ios/AppDelegates/EXAppDelegateWrapper.mm#L64. that's not necessarily to override in this call path. however when AppDelegate.swift -> ExpoAppInstance, then you are right.

alright, let's keep it as-is. it does not impact much from exposing `createRCTRootViewFactory`